### PR TITLE
BAH-1634  Upgrade OMRS version from 2.1.6 to 2.4.2 in Appointments Module

### PR DESCRIPTION
**Pull request by ayush0260**
### Summary:

This is a story. https://bahmni.atlassian.net/browse/BAH-1634
Upgraded openMRSVersion to 2.4.2

### Solution
Add missing dependencies that have been generated due to version upgrade. Modify test cases and test data to accommodate changes as a result. 

## The following changes have been made:

- udpated
openMRS v2.4.2 -> 2.5.0
rest-omod-common v2.12 -> 2.29.0
junitVersion v4.12 -> 4.13
powerMockVersion v1.6.5 -> 2.0.7
hamcrestVersion v1.3 -> 2.2
mockitoVersion v1.10.19 -> 3.5.11
modified static methods to accommodate null fields in junit 4.13
AnnotationMethodHandlerAdapter -> RequestMappingHandlerAdapter 
DefaultAnnotationHandlerMapping -> RequestMappingHandlerMapping

- added 
javax.mail v1.6.2
PowerMockIgnore for many test classes to handle warnings
packagesToScan for openmrs library in TestingApplicationContext.xml

- removed
not being used mocks in test methods in class AppointmentRecurringPatternServiceImplTest.java

### Testing
The screenshots for successful compilation have been added on the ticket.

### CODE REVIEW

File: api/pom.xml [hunk 1]
```diff
@@ -136,6 +136,11 @@
 			<artifactId>servlet-api</artifactId>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
```
Comment (angshu) [inline api/pom.xml:139]: Dont include compile time dependency. OpenMRS already bundles javax.mal https://github.com/openmrs/openmrs-core/blob/master/pom.xml#L362 

So you just need to make the scope as "provided" or "test"

File: api/src/test/java/org/openmrs/module/appointments/service/impl/AppointmentServiceDefinitionServiceImplTest.java [hunk 2]
```diff
@@ -26,6 +27,7 @@
 import static org.mockito.Mockito.when;
 import static org.powermock.api.mockito.PowerMockito.mockStatic;
 
+@PowerMockIgnore("javax.management.*")
```
Comment (angshu) [inline api/src/test/java/org/openmrs/module/appointments/service/impl/AppointmentServiceDefinitionServiceImplTest.java:30]: why is this needed?

**Review by angshu — CHANGES_REQUESTED** [review]

File: api/src/test/java/org/openmrs/module/appointments/service/impl/AppointmentServiceDefinitionServiceImplTest.java (hunk 2 above)
Comment (ayush0260) [reply api/src/test/java/org/openmrs/module/appointments/service/impl/AppointmentServiceDefinitionServiceImplTest.java:30]: PowerMockito initiates loader with classes found in this library ("javax.management.*"), so we face a linkage error when running it on the current class. 
`2022-06-10 10:41:01,526 main ERROR Could not reconfigure JMX java.lang.LinkageError: loader constraint violation: loader (instance of org/powermock/core/classloader/javassist/JavassistMockClassLoader) previously initiated loading for a different type with name "javax/management/MBeanServer"
`  

Found the solution from this link: [Mockito + PowerMock LinkageError while mocking system class](https://stackoverflow.com/questions/16520699/mockito-powermock-linkageerror-while-mocking-system-class)

File: api/pom.xml (hunk 1 above)
Comment (ayush0260) [reply api/pom.xml:139]: Updated the file

File: atom-feed/pom.xml [hunk 3]
```diff
@@ -84,7 +84,7 @@
                                             <counter>LINE</counter>
                                             <value>COVEREDRATIO</value>
                                             <!-- Lowered as PowerMockito is not covered -->
-                                            <minimum>0.43</minimum>
+                                            <minimum>0.37</minimum>
```
Comment (angshu) [inline atom-feed/pom.xml:87]: any reason we are downgrading the version?

**Review by angshu — APPROVED** [review]

File: atom-feed/pom.xml (hunk 3 above)
Comment (ayush0260) [reply atom-feed/pom.xml:87]: https://bahmni.atlassian.net/browse/BAH-1926
Created a ticket for analysis of this.
